pkg/build/api/types.go

```diff
@@ -231,6 +231,11 @@ type BuildOutput struct {
 	// a Docker image repository to push to. Failure to find the To will result in a build error.
 	To *kapi.ObjectReference `json:"to,omitempty"`
 
+	// pushSecret is the name of a Secret that would be used for setting
+	// up the authentication for executing the Docker push to authentication
+	// enabled Docker Registry (or Docker Hub).
+	PushSecret string `json:"pushSecret,omitempty"`
```

Comment:
@bparees convinced me that we don't need Name as we also not using Ref
suffix.
On Mar 24, 2015 6:09 PM, "Clayton Coleman" <notifications@github.com> wrote:

> In pkg/build/api/types.go
> <https://github.com/openshift/origin/pull/1411#discussion_r27050027>:
>
> > @@ -231,6 +231,11 @@ type BuildOutput struct {
> >  	// a Docker image repository to push to. Failure to find the To will result in a build error.
> >  	To *kapi.ObjectReference `json:"to,omitempty"`
> >
> > +	// pushSecret is the name of a Secret that would be used for setting
> > +	// up the authentication for executing the Docker push to authentication
> > +	// enabled Docker Registry (or Docker Hub).
> > +	PushSecret string `json:"pushSecret,omitempty"`
>
> Needs to be consistent with Paul's latest, which is "SecretName". You
> should not merge this until that discussion settles upstream.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/1411/files#r27050027>.
>